### PR TITLE
General updates for 1.12

### PR DESCRIPTION
**Pull request by Meronat**
This pull request covers a few things:

- Updates Sponge submodule to 1.12 for now for development reasons
- Removes achievements completely
- Updates dependencies to Minecraft/Sponge 1.12 versions
- Updates other dependencies for latest features, performance, etc
- Removes usage of all removed and some deprecated methods from Guava and Sponge
- Switch to Minecrell's [TerminalConsoleAppender](https://github.com/Minecrell/TerminalConsoleAppender) for handling our console needs. It utilizes Jline3 and JNA, could possibly switch to jansi if desired. This allows for us to simply update a library and have a lot less code to maintain. 
- A few other small things

Note: I was able to get it to run as fine as normal, except the server does not seem to generate any initial world. Doesn't seem like something my changes could have done though, but worth pointing out.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -46,6 +47,7 @@ apply plugin: 'com.github.hierynomus.license'
 apply plugin: 'com.github.johnrengelman.shadow'
 apply plugin: 'org.lanternpowered.gradle'
 apply plugin: 'org.lanternpowered.run-configs'
+apply plugin: 'de.sebastianboegl.shadow.transformer.log4j'
```
Comment (Cybermaxke) [inline build.gradle]: @Meronat There is already a transformer being applied that should handle this, it is included in the `org.lanternpowered.gradle` plugin and the transformer is applied in the `shadowJar` task.

File: src/main/java/org/lanternpowered/server/console/FullLocationPatternConverter.java [hunk 2]
```diff
@@ -23,12 +23,13 @@
...
  * THE SOFTWARE.
  */
-package org.lanternpowered.server.console.launch;
+package org.lanternpowered.server.console;
 
 import org.apache.logging.log4j.core.LogEvent;
 import org.apache.logging.log4j.core.config.plugins.Plugin;
 import org.apache.logging.log4j.core.pattern.ConverterKeys;
 import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.lanternpowered.server.game.Lantern;
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/console/FullLocationPatternConverter.java]: Unused import, as far I can tell ;)

File: src/main/resources/log4j2_indev.xml [hunk 3]
```diff
@@ -2,16 +2,16 @@
 <Configuration status="WARN" packages="org.lanternpowered.server.console.launch">
     <Appenders>
         <TerminalConsole name="Console">
-            <PatternLayout pattern="[%d{HH:mm:ss}] [%t/%level] [%logger{1}]%fqcn{ [%path]}: %msg%n"/>
```
Comment (Cybermaxke) [inline src/main/resources/log4j2_indev.xml]: @Meronat You removed the `FullLocationPatternConverter` formatting, you should readd the `%fqcn{ [%path]}`

Comment (Cybermaxke) [inline src/main/resources/log4j2_indev.xml]: Use `%loc{ [%path]}` instead, this will also work

**Comment by Meronat** [comment]
Ping me for any other desired changes.

File: build.gradle [hunk 4]
```diff
@@ -100,6 +103,7 @@ configurations {
     compile {
         extendsFrom fastutil
     }
+    log4j
```
Comment (Cybermaxke) [inline build.gradle]: @Meronat Do we actually need another sourceset to handle the `Log4j2Plugins` data file? Was it not handled properly before?

File: src/main/resources/log4j2_indev.xml [hunk 5]
```diff
@@ -2,16 +2,16 @@
...
+            <PatternLayout pattern="%highlightError{[%d{HH:mm:ss} %level] [%logger{1}] %loc{ [%path]}: %minecraftFormatting{%msg}}%n"/>
```
Comment (Cybermaxke) [inline src/main/resources/log4j2_indev.xml]: @Meronat, the space before `%loc` should be removed.

File: build.gradle [hunk 6]
```diff
@@ -110,45 +114,56 @@ dependencies {
...
     // Scripting
     compile 'org.codehaus.groovy:groovy-all:2.4.11'
 
     // Asm
     compile 'org.ow2.asm:asm-all:5.0.3'
 
+    // Configuration until Sponge includes these versions
+    compile 'ninja.leaping.configurate:configurate-hocon:3.3'
+    compile 'ninja.leaping.configurate:configurate-gson:3.3'
+    compile 'ninja.leaping.configurate:configurate-yaml:3.3'
```
Comment (Cybermaxke) [inline build.gradle]: @Meronat Configurate got updated to version 3.3 in the 1.12 branch.

File: src/main/java/org/lanternpowered/server/console/ConsoleManager.java [hunk 7]
```diff
@@ -70,51 +80,56 @@ public ConsoleManager(Logger logger, LanternScheduler scheduler, CommandManager
...
-        try {
-            reader.setHistory(new FileHistory(this.consoleHistoryFile.toFile()));
-        } catch (IOException e) {
-            this.logger.error("Error while loading the console history!", e);
-        }
+        final Terminal terminal = TerminalConsoleAppender.getTerminal();
+
+        if (terminal != null) {
+            LineReader reader = LineReaderBuilder.builder()
+                    .appName("LanternServer")
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/console/ConsoleManager.java]: Use the constant: `InternalPluginsInfo.Implementation.NAME`

**Pull request closed without merging**